### PR TITLE
py Spark Docker file

### PR DESCRIPTION
**Pull request by akbari-ali**

### CODE REVIEW

File: jupyter-images/pySpark/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,55 @@
+# Copyright (c) Jupyter Development Team.
+# Distributed under the terms of the Modified BSD License.
+ARG BASE_CONTAINER=jupyter/scipy-notebook
```
Comment (LeonardAukea) [inline jupyter-images/pySpark/Dockerfile]: @akbari-ali Maybe we should use our custom scipy image as base? But not a big deal for now

**Comment by LeonardAukea** [comment]
@the r-image is included in this pr as well. Should I close the other one and let you do the modifications here?

File: jupyter-images/pySpark/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+ENV SPARK_OPTS --driver-java-options=-Xms1024M --driver-java-options=-Xmx4096M --driver-java-options=-Dlog4j.logLevel=info
+
+USER $NB_UID
+
+# Install pyarrow
+RUN conda install --quiet -y 'pyarrow' && \
+    conda clean --all -f -y && \
+    fix-permissions $CONDA_DIR && \
+    fix-permissions /home/$NB_USER
+
```
Comment (LeonardAukea) [inline jupyter-images/pySpark/Dockerfile]: Add jupyter lab extensions as in https://github.com/volvo-cars/kubeflow-images/blob/a64fcfe622c6f1633bb8913151bdac1c0fc9b8ee/jupyter-images/scipy/Dockerfile#L20

File: jupyter-images/pySpark/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+# Spark and Mesos config
+ENV SPARK_HOME /usr/local/spark
+ENV PYTHONPATH $SPARK_HOME/python:$SPARK_HOME/python/lib/py4j-0.10.7-src.zip
+#ENV MESOS_NATIVE_LIBRARY /usr/local/lib/libmesos.so
+ENV SPARK_OPTS --driver-java-options=-Xms1024M --driver-java-options=-Xmx4096M --driver-java-options=-Dlog4j.logLevel=info
+
+USER $NB_UID
+
+# Install pyarrow
```
Comment (LeonardAukea) [inline jupyter-images/pySpark/Dockerfile]: Install dask as well

File: jupyter-images/pySpark/Dockerfile (hunk 2 above)
Comment (LeonardAukea) [reply jupyter-images/pySpark/Dockerfile]: With the addition of this https://github.com/dask/dask-labextension for dask

Comment (LeonardAukea) [reply jupyter-images/pySpark/Dockerfile]: There might be some nice ones specifc to spark as well

File: jupyter-images/R-image/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+RUN conda install --quiet --yes r-e1071
+
+ENV NB_PREFIX /
+
+CMD ["sh","-c", "jupyter lab --notebook-dir=/home/jovyan --ip=0.0.0.0 --no-browser --allow-root --port=8888 --NotebookApp.token='' --NotebookApp.password='' --NotebookApp.allow_origin='*' --NotebookApp.base_url=${NB_PREFIX}"]
+
+
+#FROM r-base
+
+#RUN apt-get update -qq && apt-get -y --no-install-recommends install \
```
Comment (LeonardAukea) [inline jupyter-images/R-image/Dockerfile]: Remove comments

File: jupyter-images/R-image/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+#   --deps TRUE \
+#    devtools \
+#    spdep
+
+
+
+
+
+
+#docker build . -t rnotebook
```
Comment (LeonardAukea) [inline jupyter-images/R-image/Dockerfile]: remove comments

File: jupyter-images/pySpark/Dockerfile (hunk 1 above)
Comment (akbari-ali) [reply jupyter-images/pySpark/Dockerfile]: Well you might not all those packages. You can do the calculation only pySpark and nothing else.

**Comment by LeonardAukea** [comment]
@akbari-ali  Planning to fix according to comments?

**Pull request merged**